docs/upgrade/upgrade-opensource/upgrade-guide-from-5.1-to-5.2/upgrade-guide-from-5.1-to-5.2-generic.rst

```diff
@@ -0,0 +1,410 @@
+--------------------------------------------------------------------------------
+
+This will cause the Scylla cluster to enable Raft and use it to consistently manage cluster-wide metadata as soon as you finish upgrading every node to the new version.
+Check the :doc:`Raft in ScyllaDB document </architecture/raft/>` to learn more.
+
+.. TODO: include enterprise versions
+
+The option is optional in 5.2 and disabled by default.
+In 5.3 it will be enabled by default, but you'll be able to disable it explicitly during upgrade if needed.
+In further versions the option will be removed and consistent cluster management will be enabled unconditionally.
+
+The option can also be enabled after the cluster is upgraded to |NEW_VERSION| (see :ref:`Enabling Raft in existing cluster <enabling-raft-existing-cluster>`).
+
+To enable the option, modify the ``scylla.yaml`` configuration file in ``/etc/scylla/`` and add the following:
+
+.. code:: yaml
+
+    consistent_cluster_management: true
+
+
+Start the node
+--------------
+
+.. code:: sh
+
+   sudo service scylla-server start
+
+Validate
+--------
+#. Check cluster status with ``nodetool status`` and make sure **all** nodes, including the one you just upgraded, are in ``UN`` status.
+#. Use ``curl -X GET "http://localhost:10000/storage_service/scylla_release_version"`` to check the ScyllaDB version. Validate that the version matches the one you upgraded to.
+#. Check scylla-server log (by ``journalctl _COMM=scylla``) and ``/var/log/syslog`` to validate there are no new errors in the log.
+#. Check again after two minutes, to validate no new issues are introduced.
+
+Once you are sure the node upgrade was successful, move to the next node in the cluster.
+
+See |Scylla_METRICS|_ for more information..
+
+After upgrading every node
+==========================
+
+The following section applies only if you enabled the ``consistent_cluster_management`` option on every node when upgrading the cluster.
+
+.. _validate-raft-setup:
+
+Validate Raft setup
+-------------------
+
+Enabling ``consistent_cluster_management`` on every node during upgrade will cause the Scylla cluster to start an additional internal procedure as soon as every node is upgraded to the new version.
+The goal of this procedure is to initialize data structures used by the Raft algorithm to consistently manage cluster-wide metadata such as table schemas.
+
+Assuming you performed the rolling upgrade procedure correctly, in particular ensuring that schema is synchronized on every step, and if there are no problems with cluster connectivity, then this follow-up internal procedure should take no longer than a few seconds to finish.
+However, the procedure requires **full cluster availability**. If an unlucky accident (e.g. a hardware problem) causes one of your nodes to fail before this procedure finishes, the procedure may get stuck. This may cause the cluster to end up in a state where schema change operations are unavailable.
```

Comment:
If we're saying that the procedure may get stuck we must be very explicit about how it may happen. A failure can be intermittent or you can lose your node completely. Let's be explicit, that the procedure may get stuck if you lose the node (at the right  moment during the procedure) forever. And point at the recovery procedure right here, since we have it for this case.

Reply:
There is a single place where the conditions for failure and the steps to handle them are described in a detailed way - the Raft document (which is linked below). I don't think we should repeat this here, it will make the thing much harder to maintain, with little value IMO.